chia/full_node/full_node.py

```diff
@@ -1153,42 +1153,36 @@ async def respond_block(
             pre_validation_results: Optional[
                 List[PreValidationResult]
             ] = await self.blockchain.pre_validate_blocks_multiprocessing([block], npc_results)
+            prevalidation_done = time.time()
             if pre_validation_results is None:
                 raise ValueError(f"Failed to validate block {header_hash} height {block.height}")
             if pre_validation_results[0].error is not None:
                 if Err(pre_validation_results[0].error) == Err.INVALID_PREV_BLOCK_HASH:
```

Comment:
The *true* case in this if-statement is always a failure. Previously, it would record the failure in `added` and `error_code` and then continue through and check it later.

This change handles the failure directly and returns. This means that all paths in the *true* block for this if-statement will return or throw. Hence the else-block is not necessary. That block can run unconditionally (which also is the actual full validation of the block, that we really need to run unconditionally unless there's an earlier error).

It's best to review this patch with white space ignored because of this indentation change.